tensorflow_addons/image/filters.py

```diff
@@ -34,82 +35,88 @@ def func2():
     return tf.cond(tf.math.greater(ma, one), func2, func1)
 
 
-@tf.function
-def mean_filter2d(image, filter_shape=(3, 3), name=None):
-    """This method performs Mean Filtering on image. Filter shape can be user
-    given.
+def _pad(image, filter_shape, mode="CONSTANT", constant_values=0):
+    """Explicitly pad a 4-D image."""
+    assert mode in ["CONSTANT", "REFLECT", "SYMMETRIC"]
+    filter_height, filter_width = filter_shape
+    pad_top = (filter_height - 1) // 2
+    pad_bottom = filter_height - 1 - pad_top
+    pad_left = (filter_width - 1) // 2
+    pad_right = filter_width - 1 - pad_left
+    paddings = [[0, 0], [pad_top, pad_bottom], [pad_left, pad_right], [0, 0]]
+    return tf.pad(image, paddings, mode=mode, constant_values=constant_values)
```

Comment:
Can we add an unit test for even shape, say (4, 4), and make sure the result is the same with scipy?

Reply:
Should we import scipy for test here?

Reply:
I could add a test case to check whether the output of explicitly zero padding is the same with the one of native "SAME" padding supported by depthwise_conv2d if we really do not want to compare the result produced by other packages.

Reply:
Hi, Tzu-Wei, perhaps we don't have to import scipy. Can we rename test_reflect_padding to test_reflect_padding_with_3x3_filter, and create an new test_reflect_padding_with_4x4_filter in the same way?

Reply:
Hi, Facai, after a rough test, I found the padding method (or the anchor of even-sized kernel) of tensorflow seems different from the one of scipy. Here is the [testing notbook](https://colab.research.google.com/drive/17tB18pyGqdlV16wsJ3YKe_mfPo4d4t7q). Take 4x4 filter for example. The explicit (zero) padding could yield the same result with implicit SAME padding supported by depthwise_conv2d. However, I have to adjust `origin` argument in [scipy's implement](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.uniform_filter.html) to (-1, -1) so that it could produce the same result. In this case, should we follow the scipy's implementation or match the implicit padding in tensorflow?

Reply:
BTW, 
scipy's reflect padding => (c b a | a b c d | d c b)
TF's reflect padding => (d c b | a b c d | c b a)
TF's reflect padding = scipy's mirror padding
scipy's reflect padding = TF's symmetric padding

Seems that they have different meaning...
https://www.tensorflow.org/api_docs/python/tf/pad

Reply:
Hi, Tzu-Wei, 
1) If it's not easy to be compatible with scipy's implementation, we can give out all details about how we calculate it in the document.
2) As for padding mode, I think [`tf.pad`](https://www.tensorflow.org/api_docs/python/tf/pad) keeps consistent with [`np.pad`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.pad.html). And similarity, we can either keep compatible with scipy, or write a detailed document.
Both are fine for me. cc @seanpmorgan Sean, what's your opinion?

Reply:
Sure :-) In either way, it should be more docs there so let me write some details in this version first.